File: Lesson/Lesson 6 Thermistor experiment/Java/Thermistor.java
import com.pi4j.wiringpi.I2C;
import com.pi4j.wiringpi.Gpio;

public class Thermistor {
    static int Buzzer = 4, led_pin = 5, value = 0, fd;

    static {
        Gpio.wiringPiSetup();
        fd = I2C.wiringPiI2CSetup(0x04);
        Gpio.pinMode(led_pin, Gpio.OUTPUT);
    }

    public static void main(String[] args){
        for ( ; ;){
            value =  I2C.wiringPiI2CReadReg16(fd, 0x10);
            if(value > 200) {
                Gpio.digitalWrite(led_pin, Gpio.HIGH);
                Gpio.digitalWrite(Buzzer, Gpio.HIGH);
                Gpio.delay(1000);
            } else {
                Gpio.digitalWrite(led_pin, Gpio.LOW);
                Gpio.digitalWrite(Buzzer, Gpio.LOW);
            }
        }

    }
}

File: Lesson/Lesson 6 Thermistor experiment/Python/Thermistor.py
import time
import smbus as smbus
import RPi.GPIO as GPIO  
import time
 
LEDPIN = 19
cyc =0.5
ADC=smbus.SMBus(1)#Declare to use I2C 1
GPIO.setmode(GPIO.BCM)
GPIO.setup(LEDPIN, GPIO.OUT)

while True:
    ADC.write_byte(0x04,0x10)#Write a byte to the slave
    val = ADC.read_word_data(0x04,0x10)
    print(val)#Raspberry Pi reads the data returned by the expansion board and prints it out  
    if val>=800:
        GPIO.output(LEDPIN, True)
        time.sleep(0.5)
    else:
        GPIO.output(LEDPIN, False)
    time.sleep(1)#Delay 1 second